proxy/src/main/java/com/wavefront/agent/PointHandlerImpl.java

```diff
@@ -172,7 +176,8 @@ public void handleBlockedPoint(String pointLine) {
     randomPostTask.incrementBlockedPoints();
   }
 
-  private static final long MILLIS_IN_YEAR = DateUtils.MILLIS_PER_DAY * 365;
+  //private static final long MILLIS_IN_YEAR = DateUtils.MILLIS_PER_DAY * 365;
+  private static final long MILLIS_DISCARD = DateUtils.MILLIS_PER_DAY * (discardPointsHours/24);
```

Comment:
This does not work (and you should write a test for this PR), the static field here would have been set during class initialization which happens before the constructor call. The field should be a private final long field that's specific to each class instantiation (and not static). You would set it every time in the constructor. The fact that you were confused that discardPointHours works is a reason why the static field should be named DISCARD_POINT_HOURS (if not DEFAULT_DISCARD_POINT_HOURS).